### PR TITLE
Algorithms on Primality Test

### PR DESCRIPTION
**Pull request by Priyangshuyogi**
## SHORT DESCRIPTION
Consist of various methods to find a number is prime or not.

## LOCATION
 I am including this in a folder named Primality Test in math folder inside Competitive programming. 

##INSIDE

All these methods are described in its perspective README.md and the implementation part is in the ScrCode.cpp file.

### CODE REVIEW

**Comment by prateekiiest** [comment]
Please specify the issue name

**Comment by Priyangshuyogi** [comment]
Issue  #185 , interesting math algo in Competitive programming.

**Comment by prateekiiest** [comment]
edit the issue name pls

**Comment by Priyangshuyogi** [comment]
Ok it is done.

**Comment by Priyangshuyogi** [comment]
@prateekiiest long time no see.

File: Competitive Coding/Math/Primality Test/Fermat Method/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Given a number n, check if it is prime or not.  This method is a probabilistic method and is based on below Fermat’s Little Theorem.
+
+Fermat's Little Theorem:
+If n is a prime number, then for every a, 1 <= a < n,
+
+a^n-1 ≡ 1 (mod n)
```
Comment (prateekiiest) [inline Competitive Coding/Math/Primality Test/Fermat Method/README.md]: add a tilde here like this `a^n-1 = 1 mod(n)`

File: Competitive Coding/Math/Primality Test/Optimized School Method/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Examples of first few prime numbers are {2, 3, 5, 7,...}
+
+Examples:
+
+Input:  n = 11
+
+Output: true
+
+Input:  n = 15
+
```
Comment (prateekiiest) [inline Competitive Coding/Math/Primality Test/Optimized School Method/README.md]: make all input / output bold and tilde for  n assignment

File: Competitive Coding/Math/Primality Test/Optimized School Method/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+A simple solution is to iterate through all numbers from 2 to n-1 and for every number check if it divides n. If we find 
+any number that divides, we return false.   Instead of checking till n,  we can check till √n because a larger factor of n 
+must be a multiple of smaller factor that has been already checked.  The algorithm can be improved further by observing 
+that all primes are of the form 6k ± 1, with the exception of 2 and 3.  This is because all integers can  be  expressed 
+as (6k + i) for some integer k and for i = ?1, 0, 1, 2, 3, or 4; 2 divides  (6k + 0),  (6k + 2),  (6k + 4); and 3 divides 
+(6k + 3).  So a more efficient method is to test if n is divisible by 2 or 3,  then to check through all the numbers of 
+form 6k ± 1.
+
+Time complexity of this solution is O(√n).
```
Comment (prateekiiest) [inline Competitive Coding/Math/Primality Test/Optimized School Method/README.md]: `O(root(n))`

**Review by prateekiiest — CHANGES_REQUESTED** [review]
Just a few comments.

looks good though ㊗️ 🎆

**Comment by Priyangshuyogi** [comment]
is it all ok now??

**Comment by baolao** [comment]
Yes thank you

File: Competitive Coding/Math/Primality Test/Fermat Method/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+Given a number n, check if it is prime or not.  This method is a probabilistic method and is based on below Fermat’s Little Theorem.
+
+Fermat's Little Theorem:
+If n is a prime number, then for every a, 1 <= a < n,
+
+a^n-1 = 1 mod (n)
```
Comment (prateekiiest) [inline Competitive Coding/Math/Primality Test/Fermat Method/README.md]: use tilde here ``

File: Competitive Coding/Math/Primality Test/Fermat Method/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ OR 
+a^n-1 % n = 1
```
Comment (prateekiiest) [inline Competitive Coding/Math/Primality Test/Fermat Method/README.md]: also here

**Comment by Priyangshuyogi** [comment]
Is it okay??

**Comment by prateekiiest** [comment]
Sorry for the so long delay.

**Pull request merged**